addons/l10n_br_pix/models/res_partner_bank.py

```diff
@@ -0,0 +1,111 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import re
+
+from odoo import models, fields, api, _
+from odoo.addons.mail.tools.mail_validation import mail_validate
+from odoo.exceptions import ValidationError
+from odoo.tools import float_repr
+
+
+class ResPartnerBank(models.Model):
+    _inherit = "res.partner.bank"
+
+    proxy_type = fields.Selection(
+        selection_add=[
+            ("br_email", "Email (BR)"),
+            ("br_cpf_cnpj", "CPF/CNPJ (BR)"),
+            ("br_mobile", "Mobile Number (BR)"),
+            ("br_random", "Random Key (BR)"),
+        ],
+        ondelete={
+            "br_email": "set default",
+            "br_cpf_cnpj": "set default",
+            "br_mobile": "set default",
+            "br_random": "set default",
+        },
+    )
+
+    @api.constrains("proxy_type", "proxy_value", "partner_id")
+    def _check_br_proxy(self):
+        for bank in self.filtered(lambda bank: bank.country_code == "BR" and bank.proxy_type != "none"):
+            if not bank.proxy_type.startswith("br_"):
```

Comment:
I do not see that logic in the other localizations.   They seem to reuse the same and the constraint seems to check if the type is within the ones for Brazil.  (we could discuss and see what is the best)

Reply:
I didn't notice multiple modules selection_add `mobile` and `email`. That works too I suppose. I did the same for Pix now, but we still have the other two Pix specific ones.